src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java

```diff
@@ -0,0 +1,459 @@
+		this.poseEstimator = new SwerveDrivePoseEstimator(this.kinematics,
+				getPigeonAngle(),
+				getSwerveModulePositions(),
+				new Pose2d(new Translation2d(), new Rotation2d()),//Limelight.getInstance().estimateLocationByVision(),
+				new MatBuilder<>(Nat.N3(), Nat.N1()).fill(RobotMap.Vision.STANDARD_DEVIATION_ODOMETRY, RobotMap.Vision.STANDARD_DEVIATION_ODOMETRY, RobotMap.Vision.STANDARD_DEVIATION_ODOMETRY),
+				new MatBuilder<>(Nat.N3(), Nat.N1()).fill(RobotMap.Vision.STANDARD_DEVIATION_VISION2D, RobotMap.Vision.STANDARD_DEVIATION_VISION2D, RobotMap.Vision.STANDARD_DEVIATION_VISION_ANGLE));
+		SmartDashboard.putData("field", getField());
+		SmartDashboard.putData("field", getField());
+	}
+	
+	
+	public static SwerveChassis getInstance() {
+		init();
+		return instance;
+	}
+	
+	
+	public static void init() {
+		if (instance == null) {
+			instance = new SwerveChassis();
+		}
+	}
+	
+	@Override
+	public void periodic() {
+		updateOdometry();
+		updatePoseEstimationLimeLight();
+		field.setRobotPose(getRobotPose());
+	}
+	
+	public void resetAll(Pose2d pose) {
+		poseEstimator.resetPosition(getPigeonAngle(), getSwerveModulePositions(), pose);
+	}
+	
+	/**
+	 * @return returns the swerve module based on its name
+	 */
+	public SwerveModule getModule(Module module) { //TODO make private
+		switch (module) {
+			case BACK_LEFT:
+				return backLeft;
+			case BACK_RIGHT:
+				return backRight;
+			case FRONT_LEFT:
+				return frontLeft;
+			case FRONT_RIGHT:
+				return frontRight;
+		}
+		return null;
+	}
+	
+	/**
+	 * stops all the modules (power(0))
+	 */
+	public void stop() {
+		frontRight.stop();
+		frontLeft.stop();
+		backRight.stop();
+		backLeft.stop();
+	}
+	
+	/**
+	 * resetting all the angle motor's encoders to 0
+	 */
+	public void resetEncodersByCalibrationRod() {
+		getModule(Module.FRONT_LEFT).resetEncoderToValue(0);
+		getModule(Module.FRONT_RIGHT).resetEncoderToValue(0);
+		getModule(Module.BACK_LEFT).resetEncoderToValue(0);
+		getModule(Module.BACK_RIGHT).resetEncoderToValue(0);
+	}
+	
+	public void resetAllEncoders() {
+		getModule(Module.FRONT_LEFT).resetEncoderByAbsoluteEncoder(Module.FRONT_LEFT);
+		getModule(Module.FRONT_RIGHT).resetEncoderByAbsoluteEncoder(Module.FRONT_RIGHT);
+		getModule(Module.BACK_LEFT).resetEncoderByAbsoluteEncoder(Module.BACK_LEFT);
+		getModule(Module.BACK_RIGHT).resetEncoderByAbsoluteEncoder(Module.BACK_RIGHT);
+		
+	}
+	
+	/**
+	 * get the absolute encoder value of a specific module
+	 */
+	public double getModuleAbsoluteEncoderValue(Module module) {
+		return getModule(module).getAbsoluteEncoderValue();
+	}
+	
+	
+	/**
+	 * all code below is self-explanatory - well, after a long time It's maybe not self-explanatory
+	 * <p>
+	 * ALL IN RADIANS, NOT DEGREES
+	 */
+	
+	public double getModuleAngle(Module module) {
+		return getModule(module).getModuleAngle();
+	}
+	
+	public boolean moduleIsAtAngle(Module module, double targetAngleInRads, double errorInRads) {
```

Comment:
I think that the 'is' should be in the begging of the function name, as it returns a boolean value